WebService/DependencyResolution.cs

```diff
@@ -90,6 +90,7 @@ private static void SetupCustomRules(ContainerBuilder builder)
             builder.RegisterType<DeviceModels>().As<IDeviceModels>().SingleInstance();
             builder.RegisterType<Services.Devices>().As<IDevices>().SingleInstance();
             builder.RegisterType<RateLimiting>().As<IRateLimiting>().SingleInstance();
+            builder.RegisterType<SimulationRunner>().As<ISimulationRunner>().SingleInstance();
```

Comment:
please add a comment explaining why this needs to be a singleton, something like "The simulation runner contains the service counters, which are read and written by multiple parts of the application, so we need to make sure there is only one instance storing that information"

Reply:
+1

Reply:
+1